docs/source/reference/api-server/examples/api-server-persistence-example.rst

```diff
@@ -0,0 +1,206 @@
+
+This example shows how a helm-deployed API server's state can be persisted using an external persistent disk.
+
+In this example, a SkyPilot API server is deployed using the Helm chart on a GKE cluster.
+This server's state is persisted with a GCP Persistent Disk, such that the server's state is not lost even if the GKE cluster is torn down.
+
+Prerequisites
+-------------
+
+* A GKE cluster with LoadBalancer or NodePort service support
+* `Helm <https://helm.sh/docs/intro/install/>`_
+* `kubectl <https://kubernetes.io/docs/tasks/tools/#kubectl>`_
+* `gcloud CLI <https://cloud.google.com/sdk/docs/install>`_
+* [Optional] Lambda Cloud credentials
+
+.. _api-server-persistence-example-create-disk:
+
+Create a persistent disk on GCP
+-------------------------------
+
+First, create a persistent disk on GCP. This disk is used to persist the API server's state.
+
+.. code-block:: bash
+
+   export ZONE=us-east5-b
+   export DISK_SIZE=10G
+   export DISK_NAME=sky-api-server-state
+   # This variable will be used later in the example.
+   export PV_CLASS_NAME=sky-api-server-pv-class
+   gcloud compute disks create $DISK_NAME --zone=$ZONE --size $DISK_SIZE
+
+Note the ``$ZONE`` variable must match the zone of the GKE cluster.
+
+.. _api-server-persistence-example-create-pv:
+
+Create a persistent volume on GKE
+---------------------------------
+
+Next, create a persistent volume on GKE that uses the persistent disk that was just created.
+
+``sky-pv.yaml``:
+
+.. code-block:: yaml
+
+    apiVersion: v1
+    kind: PersistentVolume
+    metadata:
+        name: sky-api-server-pv
+    spec:
+        storageClassName: $PV_CLASS_NAME
+        capacity:
+            storage: $DISK_SIZE
+        accessModes:
+            - ReadWriteOnce
+        csi:
+            driver: pd.csi.storage.gke.io
+            volumeHandle: projects/$PROJECT/zones/$ZONE/disks/$DISK_NAME
+            fsType: ext4
+
+Replace the variables in the above YAML.
+
+Note the ``$PROJECT`` and ``$ZONE`` variables must match the project and zone of the GKE cluster.
+In addition, ``$DISK_SIZE`` and ``$DISK_NAME`` must match the size and name of the persistent disk created on GCP.
+
+Apply the Persistent Volume to the GKE cluster.
+
+.. code-block:: bash
+
+    kubectl apply -f sky-pv.yaml
+
+
+.. _api-server-persistence-example-add-cloud-credentials:
+
+[Optional] Add lambda cloud credentials to the API server
+---------------------------------------------------------
+
+Next, add cloud credentials to the API server.
+
+In this example, the API server is configured to use Lambda Cloud credentials.
+
+To learn how to configure the credentials for another cloud provider, please refer to :ref:`sky-api-server-configure-credentials`.
+
+.. code-block:: bash
+
+    export NAMESPACE=skypilot
+    kubectl create namespace $NAMESPACE
+    kubectl create secret generic lambda-credentials \
+    --namespace $NAMESPACE \
+    --from-literal api_key=YOUR_API_KEY
```

Comment:
This section and the following sections seems not belong to this page? I would expect this page should talk more about, if the existing k8s cluster fails or terminated, how to migrate to another GKE cluster with all the states persisted. : )